### PR TITLE
Fixes to enable clang-cl compilation of compiler/runtime.

### PR DESCRIPTION
**Pull request by benvanik**
Also cleaning up some warnings found during the process.
Had to change the ukernel AVX detection to try compiling sources instead of relying on the presence of compiler flags as on MSVC/clang-cl there's only a single flag (`/arch:AVX512`) and its presence does not indicate support for subfeatures.

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp [hunk 1]
```diff
@@ -348,102 +348,6 @@ struct PropagateClonableOps : public OpRewritePattern<Op> {
   }
 };
 
-// Materialize copy-on-write (🐄) ops where required for |rootValue|.
-// Only valid in tensor/async ops - don't use with stream.cmd.*.
-static bool materializeCOW(Location loc, Value rootValue, OpBuilder &builder) {
```
Comment (ScottTodd) [inline compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp:353]: 🐄 moooo

**Review by ScottTodd — APPROVED** [review]
LGTM once CI passes. Thanks!

File: compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp (hunk 1 above)
Comment (benvanik) [reply compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp:353]: tumblr_mqq85qWOwE1rfjowdo1_500 1

File: runtime/src/iree/builtins/ukernel/arch/x86_64/CMakeLists.txt [hunk 2]
```diff
@@ -192,7 +192,8 @@ iree_select_compiler_opts(IREE_UK_COPTS_X86_64_AVX512_BASE
 iree_select_compiler_opts(IREE_UK_COPTS_X86_64_AVX512_VNNI_RELATIVE
   CLANG_OR_GCC
     "-mavx512vnni"
-  MSVC
+  CLANG_CL
+    "-mavx512vnni"
```
Comment (bjacob) [inline runtime/src/iree/builtins/ukernel/arch/x86_64/CMakeLists.txt]: Clang-CL supports such GCC-style flags ?

**Review by bjacob — APPROVED** [review]

File: runtime/src/iree/builtins/ukernel/arch/x86_64/CMakeLists.txt (hunk 2 above)
Comment (benvanik) [reply runtime/src/iree/builtins/ukernel/arch/x86_64/CMakeLists.txt]: good catch, it silently ignores this one - will remove for clarity

Comment (benvanik) [reply runtime/src/iree/builtins/ukernel/arch/x86_64/CMakeLists.txt]: I can use ` /clang:-mavx512vnni` but it's not getting passed to check_cxx_source_compiles hmm

Comment (benvanik) [reply runtime/src/iree/builtins/ukernel/arch/x86_64/CMakeLists.txt]: fixed! neato

**Pull request merged**